kolibri/core/assets/src/views/StorageNotification.vue

```diff
@@ -0,0 +1,177 @@
+        Storage notification
+      </h1>
+      <KGrid>
+        <!-- Grid Content -->
+        <KGridItem :layout12="{ span: 12 }">
+          <p>
+            {{ getMessage() }}
+          </p>
+        </KGridItem>
+
+        <!-- Grid Buttons -->
+        <KGridItem class="button-grid-item" :layout12="{ span: 12 }">
+          <KButton
+            :text="coreString('closeAction')"
+            style="margin-right: 10px"
+            appearance="flat-button"
+            :secondary="true"
+            @click="closeBanner"
+          />
+          <KButton
+            v-if="availableDownloads"
+            :text="$tr('goToDownloads')"
+            appearance="raised-button"
+            :secondary="true"
+            @click="closeBanner"
+          />
+          <KButton
+            v-if="isSuperAdmin"
+            :text="$tr('manageChannels')"
+            appearance="raised-button"
+            :secondary="true"
+            @click="manageChannel"
+          />
+        </KGridItem>
+      </KGrid>
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+
+  export default {
+    name: 'StorageNotification',
+    mixins: [commonCoreStrings],
+    data() {
+      return {
+        bannerOpened: true,
+        hasDevicePermissions: false,
+        isSuperAdmin: true,
+        availableDownloads: false,
+        learnOnlyDevice: false,
+        insufficientSpace: false,
+        resourcesRemoved: false,
+      };
+    },
+    computed: {
+      insufficientStorageNoDownloads() {
+        return (
+          (this.learnOnlyDevice && this.insufficientSpace) ||
+          (!this.hasDevicePermissions && !this.availableDownloads)
+        );
+      },
+      learnOnlyRemovedResources() {
+        return this.learnOnlyDevice && this.resourcesRemoved;
+      },
+      availableDownload() {
+        return !this.hasDevicePermissions && this.availableDownloads && !this.learnOnlyDevice;
+      },
+    },
+    created() {
+      document.addEventListener('focusin', this.focusChange);
+    },
+    beforeDestroy() {
+      document.removeEventListener('focusin', this.focusChange);
+    },
+    methods: {
+      getMessage() {
+        let message = '';
+        if (this.isSuperAdmin) {
+          message = this.$tr('superAdminMessage');
+        } else if (this.insufficientStorageNoDownloads) {
+          message = this.$tr('insufficientStorageNoDownloads');
+        } else if (this.learnOnlyRemovedResources) {
+          message = this.$tr('resourcesRemoved');
+        } else if (this.availableDownload) {
+          message = this.$tr('insufficientStorageAvailableDownloads');
+        }
+        return message;
+      },
+      closeBanner() {
+        this.bannerOpened = false;
+        if (this.previouslyFocusedElement) {
+          this.previouslyFocusedElement.focus();
+        }
+      },
+      manageChannel() {
+        this.$router.push('/');
```

Comment:
Let's remember to follow up on the cross-plugin routing after my android PR is merge! I think some of that can be repurposed here if needed